core/contract.go

```diff
@@ -67,83 +120,80 @@ func (c *Contract) UpdateNonce(nonce *felt.Felt) error {
 }
 
 // ClassHash returns hash of the class that this contract instantiates.
-func (c *Contract) ClassHash() (classHash *felt.Felt, err error) {
-	key := db.ContractClassHash.Key(c.Address.Marshal())
-	err = c.txn.Get(key, func(val []byte) error {
-		classHash = new(felt.Felt)
-		classHash.SetBytes(val)
-		return nil
-	})
-	return
+func (c *Contract) ClassHash() (*felt.Felt, error) {
+	return classHash(c.Address, c.txn)
 }
 
-// Storage returns the [core.Trie] that represents the
-// storage of the contract.
-func (c *Contract) Storage() (*trie.Trie, error) {
-	addrBytes := c.Address.Marshal()
-	var contractRootKey *bitset.BitSet
-
-	if err := c.txn.Get(db.ContractRootKey.Key(addrBytes), func(val []byte) error {
-		contractRootKey = new(bitset.BitSet)
-		return contractRootKey.UnmarshalBinary(val)
-	}); err != nil && !errors.Is(err, db.ErrKeyNotFound) {
-		// Don't continue normal operation with arbitrary
-		// database error.
-		return nil, err
-	}
-	trieTxn := NewTransactionStorage(c.txn, db.ContractStorage.Key(addrBytes))
-	return trie.NewTrie(trieTxn, contractStorageTrieHeight, contractRootKey), nil
-}
-
-// StorageRoot returns the root of the contract storage.
-func (c *Contract) StorageRoot() (*felt.Felt, error) {
-	if storage, err := c.Storage(); err != nil {
+// Root returns the root of the contract storage.
+func (c *Contract) Root() (*felt.Felt, error) {
+	if cStorage, err := storage(c.Address, c.txn); err != nil {
 		return nil, err
 	} else {
-		return storage.Root()
+		return cStorage.Root()
 	}
 }
 
-// UpdateStorage applies a change-set to the contract storage.
-func (c *Contract) UpdateStorage(diff []StorageDiff) error {
-	storage, err := c.Storage()
+// Update applies a change-set to the contract storage.
+func (c *Contract) Update(diff []StorageDiff) error {
```

Comment:
I thought UpdateStorage made more sense because we have UpdateNonce as well.

Reply:
I have changed `Update` to `UpdateStorage` and `Get` to `Storage`